src/overlays/actors/ovl_kaleido_scope/z_kaleido_map_PAL.c

```diff
@@ -1,5 +1,731 @@
+                    Audio_PlaySoundGeneral(NA_SE_SY_CURSOR, &D_801333D4, 4, &D_801333E0, &D_801333E0, &D_801333E8);
+                    D_8082A6D4 = 0;
+                }
+            }
+        }
+
+        if (pauseCtx->worldMapPoints[pauseCtx->cursorPoint[PAUSE_WORLD_MAP]] == 0) {
+            pauseCtx->cursorItem[PAUSE_MAP] = PAUSE_ITEM_NONE;
+        }
+
+        if (oldCursorPoint != pauseCtx->cursorPoint[PAUSE_WORLD_MAP]) {
+            Audio_PlaySoundGeneral(NA_SE_SY_CURSOR, &D_801333D4, 4, &D_801333E0, &D_801333E0, &D_801333E8);
+        }
+    }
+
+    gDPPipeSync(POLY_OPA_DISP++);
+
+    if (HREG(15) == 0) {
+        gDPSetTextureFilter(POLY_OPA_DISP++, G_TF_POINT);
+
+        gDPLoadTLUT_pal256(POLY_OPA_DISP++, gWorldMapImageTLUT);
+        gDPSetTextureLUT(POLY_OPA_DISP++, G_TT_RGBA16);
+
+        gDPSetPrimColor(POLY_OPA_DISP++, 0, 0, 255, 255, 255, pauseCtx->alpha);
+        gSPVertex(POLY_OPA_DISP++, &pauseCtx->mapPageVtx[188], 32, 0);
+
+        for (j = t = i = 0; i < 8; i++, t++, j += 4) {
+            gDPLoadTextureBlock(POLY_OPA_DISP++, (u8*)gWorldMapImageTex + t * 216 * 9, G_IM_FMT_CI, G_IM_SIZ_8b, 216, 9,
+                                0, G_TX_WRAP | G_TX_NOMIRROR, G_TX_WRAP | G_TX_NOMIRROR, G_TX_NOMASK, G_TX_NOMASK,
+                                G_TX_NOLOD, G_TX_NOLOD);
+
+            gSP1Quadrangle(POLY_OPA_DISP++, j, j + 2, j + 3, j + 1, 0);
+
+            if (1) {}
+        }
+
+        gSPVertex(POLY_OPA_DISP++, &pauseCtx->mapPageVtx[220], 28, 0);
+
+        for (j = i = 0; i < 6; i++, t++, j += 4) {
+            gDPLoadTextureBlock(POLY_OPA_DISP++, (u8*)gWorldMapImageTex + t * 216 * 9, G_IM_FMT_CI, G_IM_SIZ_8b, 216, 9,
+                                0, G_TX_WRAP | G_TX_NOMIRROR, G_TX_WRAP | G_TX_NOMIRROR, G_TX_NOMASK, G_TX_NOMASK,
+                                G_TX_NOLOD, G_TX_NOLOD);
+
+            gSP1Quadrangle(POLY_OPA_DISP++, j, j + 2, j + 3, j + 1, 0);
+        }
+
+        gDPLoadTextureBlock(POLY_OPA_DISP++, (u8*)gWorldMapImageTex + t * 216 * 9, G_IM_FMT_CI, G_IM_SIZ_8b, 216, 2, 0,
+                            G_TX_WRAP | G_TX_NOMIRROR, G_TX_WRAP | G_TX_NOMIRROR, G_TX_NOMASK, G_TX_NOMASK, G_TX_NOLOD,
+                            G_TX_NOLOD);
+
+        gSP1Quadrangle(POLY_OPA_DISP++, j, j + 2, j + 3, j + 1, 0);
+    } else if (HREG(15) == 1) {
+        Gfx* sp1CC = POLY_OPA_DISP;
+        void* mapImage = gWorldMapImageTex;
+
+        // gSPLoadUcodeL(sp1CC++, rspS2DEX)?
+        gSPLoadUcodeEx(sp1CC++, OS_K0_TO_PHYSICAL(D_80113070), OS_K0_TO_PHYSICAL(D_801579A0), 0x800);
+
+        func_8009638C(&sp1CC, mapImage, gWorldMapImageTLUT, 216, 128, G_IM_FMT_CI, G_IM_SIZ_8b, 0x8000, 256,
+                      HREG(13) / 100.0f, HREG(14) / 100.0f);
+
+        // gSPLoadUcode(sp1CC++, SysUcode_GetUCode(), SysUcode_GetUCodeData())?
+        gSPLoadUcodeEx(sp1CC++, SysUcode_GetUCode(), SysUcode_GetUCodeData(), 0x800);
```

Comment:
do these work?

Reply:
They should work, but we don't have the necessary defines/symbols yet. The first one (`rspS2DEX` stuff) will need the s2dex part of the rsp bins split, identified and named, while the second one will need a certain define (`SP_UCODE_DATA_SIZE`) for the size of ucode data which we don't have yet (see `ucode.h` in sm64 decomp for an example of what we will need in the future).

FYI I already left the same comments in the only other place where s2dex is used in the game ie. background rendering in `z_room.c`